src/python/pants/option/global_options.py

```diff
@@ -252,82 +308,94 @@ def from_options(
         store_rpc_concurrency = cast(int, bootstrap_options.remote_store_rpc_concurrency)
         cache_rpc_concurrency = cast(int, bootstrap_options.remote_cache_rpc_concurrency)
         execution_rpc_concurrency = cast(int, bootstrap_options.remote_execution_rpc_concurrency)
+        return cls(
+            execution=execution,
+            cache_read=cache_read,
+            cache_write=cache_write,
+            instance_name=instance_name,
+            store_address=cls._normalize_address(store_address),
+            execution_address=cls._normalize_address(execution_address),
+            store_headers=store_headers,
+            execution_headers=execution_headers,
+            parallelism=parallelism,
+            store_rpc_concurrency=store_rpc_concurrency,
+            cache_rpc_concurrency=cache_rpc_concurrency,
+            execution_rpc_concurrency=execution_rpc_concurrency,
+        )
 
-        if bootstrap_options.remote_oauth_bearer_token_path:
-            oauth_token = (
-                Path(bootstrap_options.remote_oauth_bearer_token_path).resolve().read_text().strip()
+    @classmethod
+    def _use_auth_plugin(
+        cls,
+        bootstrap_options: OptionValueContainer,
+        full_options: Options,
+        env: CompleteEnvironment,
+        prior_result: AuthPluginResult | None,
+    ) -> tuple[DynamicRemoteOptions, AuthPluginResult | None]:
+        auth_plugin_result: AuthPluginResult | None = None
+        if ":" not in bootstrap_options.remote_auth_plugin:
+            raise OptionsError(
+                "Invalid value for `[GLOBAL].remote_auth_plugin`: "
+                f"{bootstrap_options.remote_auth_plugin}. Please use the format "
+                f"`path.to.module:my_func`."
             )
-            if set(oauth_token).intersection({"\n", "\r"}):
-                raise OptionsError(
-                    f"OAuth bearer token path {bootstrap_options.remote_oauth_bearer_token_path} "
-                    "must not contain multiple lines."
-                )
-            token_header = {"authorization": f"Bearer {oauth_token}"}
-            execution_headers.update(token_header)
-            store_headers.update(token_header)
+        execution = cast(bool, bootstrap_options.remote_execution)
```

Comment:
These casts are duplicative of the casts in the OAuth bearer token function. Maybe have a common function to extract to an intermediate dataclass?

Reply:
As mentioned in the PR description, I think having some code duplication is ok.
it makes it easier to understand and maintain, the duplicated code is trivial.
I don't see a real benefit that we would get by introducing a data class to reduce the duplicated code.

Reply:
In general I'd agree with Tom, but I think that it's maybe not necessary here since we're constructed a dataclass which provides the type safety of not being able to forget an option.